### DISCUSSION TITLE
PAGViewer 能否增加区间循环播放呢

### DISCUSSION THREAD

**Discussion by thinylim** [Tencent/libpag#1644] [Ideas] [answered]
就是指定播放开始帧和结束帧，并且可以循环播放，看论坛里，大家比较多是需要片段式循环播放的，这样可以预览循环播放效果，希望考虑一下

**Comment by kevingpqi123** [comment]
如果需要区间播放，可以到 demo 工程查看相关范例
pag 支持所有原子素材的自由组合，你所说的区间播放只是一个很小的场景，我们更推荐在 PAGComposition 层面去实现，而不是在 PAGView 层面

**Reply by thinylim** [reply]
其实我想要的是直接在Viewer工具上能看到效果，不用到代码接入后，这样可以跟动效沟通导出的问题
另外，“区间播放，可以到 demo 工程查看相关范例” 这个我真没找到哪里的demo有这个范例，起码android的没有，demo都是直接循环播放一个pag，简单的不能再简单的范例方式了，没参考价值，很多接口对于SDK开发人员来说是很简单，觉得不用提供使用方式，如果从接入者来考虑，还是提供一个完整的代码示例会更直接，论坛也很多人问循环播放的吧，问了还有人问，这证明大家会更多的遇到这种场景，demo为什么不考虑直接给个完整的示例呢

**Reply by kevingpqi123** [reply, marked as answer]
两个问题：
1、桌面预览工具 PAGViewer 仅仅是预览效果和查看性能信息使用的，不是所有的功能都要放在上面，常用的一些使用方法都是在 pag 的 demo 工程中
2、既然论坛中有人问，肯定有相关答案，请仔细阅读：https://github.com/Tencent/libpag/discussions/1417
如果你不知道官方 demo 在哪里，请阅读官网的相关文档（  https://pag.art/docs/pad-demo-download.html ） 或者 github 的 README （ https://github.com/Tencent/libpag ）
Example
Check out the following projects to learn how to use the APIs of libpag:

https://github.com/libpag/pag-ios
https://github.com/libpag/pag-android
https://github.com/libpag/pag-web


<img width="584" alt="企业微信截图_538a58c4-00a8-4f66-9fee-3bc30c2880e6" src="https://github.com/Tencent/libpag/assets/16609221/f627bbb8-0702-44d8-a90b-9f33ca65cbc0">
